src/draw/DefaultDrawer.js

```diff
@@ -687,6 +733,48 @@ class DefaultDrawer {
       .target(() => targetAnchor);
   }
 
+  /**
+   * Compute a component's height and store it in it's drawOptions.
+   *
+   * @param {Node} component - The component to get the height for.
+   * @returns {number} - The height.
+   */
+  getComponentHeight(component) {
+    if (component.id === '__shadowRoot') {
+      return 0;
+    }
+    const containerSpacing = this.minHeight + this.padding + this.margin;
+
+    const childHeights = component.children
+      ? component.children.map(({ y1 }) => y1 + containerSpacing)
+      : [0];
+
+    component.data.drawOption.height = (Math.max(
+      this.minHeight + (component.data.definition.isContainer * containerSpacing),
+      ...childHeights,
+    ));
+
+    return component.data.drawOption.height;
+  }
+
+  /**
+   * Compute a component's width and store it in it's drawOptions.
```

Comment:
Compute the component's width then store it in its drawOptions.

Reply:
done